kernel/src/main/java/org/kframework/parser/inner/ParseInModule.java

```diff
@@ -228,10 +228,7 @@ public Scanner getScanner() {
             parseStringTerm(String input, Sort startSymbol, Scanner scanner, Source source, int startLine, int startColumn, boolean inferSortChecks, boolean isAnywhere) {
         scanner = getGrammar(scanner);
 
-        long start = 0;
-        if (profileRules) {
-            start = System.nanoTime();
-        }
+        long start = System.currentTimeMillis(), endParse = 0, startTypeInf = 0, endTypeInf = 0;
```

Comment:
I don't understand why this is being changed. Not only are we reducing the precision by a factor of 1000, you are now also calling the potentially somewhat expensive function even when its value doesn't matter.

Reply:
According to this article: https://www.geeksforgeeks.org/java-system-nanotime-vs-system-currenttimemillis/
`System.nanoTime()`
- may not be thread safe (although I've read that might not be the case with modern java versions)
- `currentTimeMillis()` executes faster than `nanoTime()` 
- the precision is irrelevant. I did some tests with nanotime and the time variation for the same rule can be 30ms. Considering that we are running multi threaded, cache misses are non-deterministic, opportunistic CPU boost... this variation is understandable. So milliseconds is completely fine in our case. We are looking for big outliers (100ms or more).

I don't feel strongly about it, I can change it back to `nanoTime()`. But according to my research, `currentTimeMillis()` is a better fit for our case.

Reply:
Alright, this is fine then.

Reply:
We should still only call the function if we are actually profiling though.